clients/spark/core/src/test/scala/io/treeverse/clients/GarbageCollectorSpec.scala

```diff
@@ -1,12 +1,29 @@
 package io.treeverse.clients
 
+import scala.collection.JavaConverters._
 import org.scalatest._
 import matchers.should._
 import funspec._
 
+import io.treeverse.lakefs.catalog
+
+import org.apache.commons.io.FileUtils
 import org.apache.spark.{HashPartitioner, SparkConf}
 import org.apache.spark.sql.{Dataset, SparkSession}
-import io.treeverse.lakefs.catalog
+
+import org.json4s._
+import org.json4s.native.JsonMethods
+
+import java.nio.file.{Files, Path, Paths}
+
+trait TempDirectory {
+  def withTempDirectory(testMethod: (Path) => Any) {
+    val tempDir = Files.createTempDirectory("test-gc")
+
+    testMethod(tempDir)
+    FileUtils.deleteDirectory(tempDir.toFile)
```

Comment:
What happens if an exception is thrown in the test? Will the file still be deleted?

Reply:
True.  Fixed to delete.